declarative/tests/http/src/test/java/io/helidon/declarative/tests/http/DeclarativeHttpTest.java

```diff
@@ -0,0 +1,136 @@
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+import io.helidon.http.HttpException;
+import io.helidon.http.Status;
+import io.helidon.service.registry.Lookup;
+import io.helidon.service.registry.Qualifier;
+import io.helidon.service.registry.ServiceRegistry;
+import io.helidon.webclient.api.RestClient;
+import io.helidon.webclient.http1.Http1Client;
+import io.helidon.webserver.testing.junit5.ServerTest;
+
+import jakarta.json.Json;
+import jakarta.json.JsonBuilderFactory;
+import jakarta.json.JsonObject;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ServerTest
+class DeclarativeHttpTest {
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Map.of());
+
+    private final Http1Client client;
+    private final ServiceRegistry registry;
+    private final URI serverUri;
+
+    protected DeclarativeHttpTest(Http1Client client,
+                                  ServiceRegistry registry,
+                                  URI serverUri) {
+        this.client = client;
+        this.registry = registry;
+        this.serverUri = serverUri;
+    }
+
+    @Test
+    void testRootRoute() {
+        var response = client.get("/greet").request(JsonObject.class);
+
+        assertThat(response.status(), is(Status.OK_200));
+        JsonObject json = response.entity();
+        assertThat(json.getString("message"), is("Hello World!"));
+    }
+
+    @Test
+    void testHealthObserver() {
+        var response = client.get("/observe/health").request(String.class);
+        assertThat(response.status(), is(Status.NO_CONTENT_204));
+    }
+
+    @Test
+    void testDeadlockHealthCheck() {
+        var response = client.get("/observe/health/live/deadlock").request(String.class);
+        assertThat(response.status(), is(Status.NO_CONTENT_204));
+    }
+
+    @Test
+    void testMetricsObserver() {
+        var response = client.get("/observe/metrics").request(String.class);
+        assertThat(response.status(), is(Status.OK_200));
+    }
+
+    @Test
+    void testErrorHandler() {
+        JsonObject badEntity = JSON.createObjectBuilder().build();
+
+        var response = client.put("/greet/greeting").submit(badEntity, JsonObject.class);
+        assertThat(response.status(), is(Status.BAD_REQUEST_400));
+        JsonObject entity = response.entity();
+        assertThat(entity.getString("error"), is("No greeting provided"));
+    }
+
+    @Test
+    void testTypedClient() {
+        GreetEndpointClient typedClient = registry.get(Lookup.builder()
+                                                               .addContract(GreetEndpointClient.class)
+                                                               .addQualifier(Qualifier.create(RestClient.Client.class))
+                                                               .build());
```

Comment:
Should there be a registry getter with a contract and zero o more qualifiers to avoid having to build a Lookup instance?

Reply:
I would like to add such a method to `Services`. Enhancement for next time.